client/rpcserver/handlers.go

```diff
@@ -228,6 +231,26 @@ func handlePreRegister(s *RPCServer, req *msgjson.Message) *msgjson.ResponsePayl
 	return createResponse(req.Route, res, nil)
 }
 
+// handleRegister handles requests for register. *msgjson.ResponsePayload.Error
+// is empty if successful.
+func handleRegister(s *RPCServer, req *msgjson.Message) *msgjson.ResponsePayload {
+	form := new(core.Registration)
+	err := json.Unmarshal(req.Payload, form)
+	if err != nil {
+		resErr := &msgjson.Error{Code: msgjson.RPCParseError, Message: "unable to unmarshal request"}
+		return createResponse(req.Route, nil, resErr)
+	}
+	err = s.core.Register(form)
+	if err != nil {
+		resErr := &msgjson.Error{Code: msgjson.RPCRegisterError, Message: err.Error()}
+		return createResponse(req.Route, nil, resErr)
+	}
+
+	resp := fmt.Sprintf(feePaidStr, form.Fee)
```

Comment:
I'm a little uncomfortable allowing the user to mess this up and pay the wrong fee.  To make this impossible and eliminate one CLI argument, I'd be fine with a preregister call to get the fee first.

Reply:
The wrong fee is met with an error.

The thing with automating that is, the server could then slip in any fee they wanted. If the client has it in their wallet, it would pay. Of course this would be suicide for the DEX, but it would make for a great exit scam. If I'm wrong about this scenario, let me know.

Reply:
That's a good point.

Reply:
Well, how about it checks preregister against the input fee to prevent typos from causing trouble.